indico/modules/events/management/forms.py

```diff
@@ -270,6 +268,20 @@ def _init_visibility(self, event):
             self.visibility.warning = _("This event's visibility is currently limited by that of '{}'.").format(
                 real_horizon.title)
 
+    def validate_permissions(self, field):
+        event = self.event
+        for idx, value in enumerate(field.data):
+            principal_fossil = value[0]
+            permissions = value[1]
+            principal = principal_from_fossil(principal_fossil, allow_emails=True, allow_networks=True, event=event)
+            if isinstance(principal, IPNetworkGroup):
+                if set(permissions) - {READ_ACCESS_PERMISSION}:
+                    msg = _('IP networks cannot have management permissions: {}')
+                    raise ValidationError(msg.format(principal.name))
+            if FULL_ACCESS_PERMISSION in permissions and len(permissions) != 1:
+                # when full access permission is set, discard rest of permissions
+                field.data[idx][1] = [FULL_ACCESS_PERMISSION]
```

Comment:
You can update the list in-place:

    permissions[:] = [FULL_ACCESS_PERMISSION]